src/components/Filters/AggregationItem/AggregationItem.js

```diff
@@ -0,0 +1,126 @@
+import PropTypes from 'prop-types';
+import { useSelector, useDispatch } from 'react-redux';
+import { FormattedNumber } from 'react-intl';
+import { filterPatch, SLUG_SEPARATOR } from '../../../constants';
+import { coalesce, sanitizeHtmlId } from '../../../utils';
+import { arrayEquals } from '../../../utils/compare';
+import { replaceFilters, toggleFilter } from '../../../actions/filter';
+import { getUpdatedFilters } from '../../../utils/filters';
+import { selectAggsState } from '../../../reducers/aggs/selectors';
+import { selectQueryState } from '../../../reducers/query/selectors';
+
+const appliedFilters = ({ fieldName, item, aggs, filters }) => {
+  // We should find the parent
+  // determine if the other siblings are already checked
+  // check the parent only, and uncheck the rest so that the fake check
+  // will take affect
+  const [parentFilter, childFilter] = item.key.split(SLUG_SEPARATOR);
+  /* eslint-disable no-unexpected-multiline */
+  // TODO: reformat to not need the unexpected multiline.
+  const subItems = aggs
+    .find((agg) => agg.key === parentFilter)
+    ['sub_' + fieldName + '.raw'].buckets.map((agg) => agg.key)
+    .sort();
+  /* eslint-enable no-unexpected-multiline */
+
+  const parentKey = parentFilter + SLUG_SEPARATOR;
+  const selectedFilters = filters
+    .filter((filter) => filter.indexOf(parentKey) > -1)
+    .map((filter) => filter.replace(parentKey, ''));
+  selectedFilters.push(childFilter);
+
+  selectedFilters.sort();
+
+  if (arrayEquals(selectedFilters, subItems)) {
+    // remove subitems, add parent filter
+    return filters
+      .filter((filter) => filter.indexOf(parentKey) === -1)
+      .concat(parentFilter);
+  } else {
+    // just add the single filter and apply filters
+    return filters.concat(item.key);
+  }
+};
+
+const AggregationItem = ({ fieldName, item }) => {
+  const aggsState = useSelector(selectAggsState);
+  const queryState = useSelector(selectQueryState);
+  const dispatch = useDispatch();
+
+  const aggs = coalesce(aggsState, fieldName, []);
+  const filters = coalesce(queryState, fieldName, []);
+  const active =
```

Comment:
booleans should have is `is` or `has` in front of them to make it clear when they're used.

https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/boolean-prop-naming.md

Reply:
I usually do it for functions and methods, but have updated here accordingly.